### DISCUSSION TITLE
Why do we need a callback with Lambda functions?

### DISCUSSION THREAD

**Discussion by cosbgn** [vercel/vercel#4350] [General]
Hello, I'm using standard lambas setting: 
```
"NODEJS_AWS_HANDLER_NAME": "handler"
"NODEJS_HELPERS":"0"
```
A simple function which works fine looks like this:

```
"use strict";
export async function handler(event, context, cb) {
	const response = {
	  statusCode: 200,
	  body: JSON.stringify({ message: 'hello world' })
	}
	cb(null, response)
}
```

However with async we don't really need the callback, so this should also work:

```
"use strict";
export async function handler(event, context) {
	const response = {
	  statusCode: 200,
	  body: JSON.stringify({ message: 'hello world' })
	}
	return response
}
```

However it fails. Any ideas why?

**Comment by styfle** [comment]
I don't know think the raw AWS API works with ESM since they don't transpile.

Try using CJS instead like this: https://github.com/zeit/now/blob/master/packages/now-node/test/fixtures/25-aws-api/index.js

**Comment by cosbgn** [comment]
Hi I tried this:

```
exports.handler = async function () {
	return {
	  statusCode: 200,
	  headers: {},
	  body: JSON.stringify({ text: 'aws-api-root:RANDOMNESS_PLACEHOLDER' }),
	};
};
```
And on `vc dev` it still doesn't work.

**Reply by cosbgn** [reply]
Also the example from [the docs](https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/aws-lambda-api) doesn't work

**Reply by styfle** [reply]
You're right, it doesn't work with dev right now, only real deployments.

We'll make sure this works correctly in dev, thanks!
